src/Umbraco.Web.UI.Client/src/common/services/angularhelper.service.ts

```diff
@@ -1,191 +1,134 @@
-                });
-            }
-
-            for (var p in scope) {
-
-                if (_.isObject(scope[p]) && p !== "this" && p.substr(0, 1) !== "$") {
-                    //get the keys of the property names for the current property
-                    var props = _.keys(scope[p]);
-                    //if the length isn't correct, try the next prop
-                    if (props.length < requiredFormProps.length) {
-                        continue;
-                    }
-
-                    //ensure that every required property name exists on the current scope property
-                    var containProperty = propertyExists(props);
-
-                    if (containProperty) {
-                        form = scope[p];
-                        break;
-                    }
-                }
-            }
-
-            return form;
-        },
-
-        /**
-         * @ngdoc function
-         * @name validateHasForm
-         * @methodOf umbraco.services.angularHelper
-         * @function
-         *
-         * @description
-         * This will validate that the current scope has an assigned form object, if it doesn't an exception is thrown, if
-         * it does we return the form object.
-         */
-        getRequiredCurrentForm: function (scope) {
-            var currentForm = this.getCurrentForm(scope);
-            if (!currentForm || !currentForm.$name) {
-                throw "The current scope requires a current form object (or ng-form) with a name assigned to it";
-            }
-            return currentForm;
-        },
-
-        /**
-         * @ngdoc function
-         * @name getNullForm
-         * @methodOf umbraco.services.angularHelper
-         * @function
-         *
-         * @description
-         * Returns a null angular FormController, mostly for use in unit tests
-         *      NOTE: This is actually the same construct as angular uses internally for creating a null form but they don't expose
-         *          any of this publicly to us, so we need to create our own.
-         *      NOTE: The properties has been added to the null form because we use them to get a form on a scope.
-         *
-         * @param {string} formName The form name to assign
-         */
-        getNullForm: function (formName) {
-            return {
-                $error: {},
-                $dirty: false,
-                $pristine: true,
-                $valid: true,
-                $submitted: false,
-                $pending: undefined,
-                $addControl: angular.noop,
-                $removeControl: angular.noop,
-                $setValidity: angular.noop,
-                $setDirty: angular.noop,
-                $setPristine: angular.noop,
-                $name: formName
-            };
-        }
-    };
-}
-angular.module('umbraco.services').factory('angularHelper', angularHelper);
+/// <reference path="../definitions/global.d.ts" />
+
+module umbraco.services {

```

Comment:
I'm defo no TS dev (though very keen on learning) but shouldn't this be `namespace` instead of `module` like the other files changed here?

Reply:
Good spot. Namespace is the one I need. Will update